apps/dav/lib/UserMigration/CalendarMigrator.php

```
@@ -0,0 +1,451 @@
+					$componentTimezoneIds[] = $timezoneId;
+				}
+			}
+		}
+
+		$calendarTimezoneMap = $this->getCalendarTimezones($vCalendar);
+
+		return array_values(array_filter(array_map(
+			fn (string $timezoneId) => $calendarTimezoneMap[$timezoneId],
+			$componentTimezoneIds,
+		)));
+	}
+
+	private function sanitizeComponent(VObjectComponent $component): VObjectComponent {
+		// Operate on the component clone to prevent mutation of the original
+		$componentClone = clone $component;
+
+		// Remove RSVP parameters to prevent automatically sending invitation emails to attendees on import
+		foreach ($componentClone->children() as $child) {
+			if (
+				$child->name === 'ATTENDEE'
+				&& isset($child->parameters['RSVP'])
+			) {
+				unset($child->parameters['RSVP']);
+			}
+		}
+
+		return $componentClone;
+	}
+
+	/**
+	 * @return VObjectComponent[]
+	 */
+	private function getRequiredImportComponents(VCalendar $vCalendar, VObjectComponent $component): array {
+		$component = $this->sanitizeComponent($component);
+		/** @var array<int, VTimeZone> $timezoneComponents */
+		$timezoneComponents = $this->getTimezonesForComponent($vCalendar, $component);
+		return [
+			...$timezoneComponents,
+			$component,
+		];
+	}
+
+	private function initCalendarObject(): VCalendar {
+		$vCalendarObject = new VCalendar();
+		$vCalendarObject->PRODID = $this->sabreDavServer::$exposeVersion
+			? '-//SabreDAV//SabreDAV ' . SabreDavVersion::VERSION . '//EN'
+			: '-//SabreDAV//SabreDAV//EN';
+		return $vCalendarObject;
+	}
+
+	private function importCalendarObject(int $calendarId, VCalendar $vCalendarObject): void {
+		try {
+			$this->calDavBackend->createCalendarObject(
+				$calendarId,
+				UUIDUtil::getUUID() . CalendarMigrator::FILENAME_EXT,
+				$vCalendarObject->serialize(),
+				CalDavBackend::CALENDAR_TYPE_CALENDAR,
+			);
+		} catch (BadRequest $e) {
+			// Rollback creation of calendar on error
+			$this->calDavBackend->deleteCalendar($calendarId, true);
```

Please report this error in the output.